src/Microsoft.ML.Api/PredictionEngine.cs

```diff
@@ -189,19 +211,24 @@ public TDst Predict(TSrc example)
             return result;
         }
 
+        public void ExtractValues(TSrc example) => _inputRow.ExtractValues(example);
+
+        public void FillValues(TDst prediction) => _outputRow.FillValues(prediction);
+
         /// <summary>
         /// Run prediction pipeline on one example.
         /// </summary>
         /// <param name="example">The example to run on.</param>
         /// <param name="prediction">The object to store the prediction in. If it's <c>null</c>, a new one will be created, otherwise the old one
         /// is reused.</param>
-        public void Predict(TSrc example, ref TDst prediction)
+        public virtual void Predict(TSrc example, ref TDst prediction)
```

Comment:
>virtual [](start = 15, length = 7)

If 2 existing implementations are dissimilar, what's the value in having a virtual method? Make it `abstract`